sw/device/silicon_creator/lib/base/mock_abs_mmio.h

```diff
@@ -14,18 +14,16 @@ namespace internal {
 /**
  * Mock class for abs_mmio.c.
  */
-class MockAbsMmio {
+class MockAbsMmio : public GlobalMock<MockAbsMmio> {
```

Comment:
I had no idea you could do this.  Somehow, this feels like a circular reference: `MockAbsMmio` derives from `GloblaMock` which is parameterized with `MockAbsMmio`.

Its perfectly clear in the `GlobalMock` implementation what is happening... just feels a bit strange to see a derived class mention itself in the base class.

Reply:
Thanks for the review @cfrantz! This pattern is called the "curiously recurring template pattern" (CRTP). You can read more about it [here](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern).